noxfile.py

```diff
@@ -0,0 +1,279 @@
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+def lint(session: "nox.session") -> None:
+    session.install("flake8")
+    if platform.python_implementation() == "CPython":
+        session.install("black")
+    if session.posargs:
+        args = session.posargs
+    else:
+        args = DEFAULT_TEST_DIRECTORIES
+    session.run("flake8", *args)
+
+    if platform.python_implementation() == "CPython":
+        if session.posargs:
+            session.run("python", "-m", "black", *session.posargs)
+        else:
+            session.run(
+                "python", "-m", "black", "--diff", "--check", *BLACK_CONFORM_FILES
+            )
+            session.run("python", "-m", "black", "--diff", *DEFAULT_TEST_DIRECTORIES)
+    else:
+        session.log(
+            f"Skip black because of platform {platform.python_implementation()}."
+        )
+
+
+@nox.session
+def black(session: "nox.session") -> None:
+    session.install("black")
+    if session.posargs:
+        args = session.posargs
+    else:
+        raise RuntimeError("Please add the files to format as argument.")
+    session.run("python", "-m", "black", *args)
+
+
+@nox.session
+def doc(session: "nox.session") -> None:
+    session.install(
+        "sphinx",
+        "sphinx_rtd_theme",
+        "sphinxcontrib-autoprogram==0.1.5 ; python_version=='3.6'",
+        "sphinxcontrib-autoprogram>=0.1.5 ; python_version>='3.7'",
+    )
+    session.install("-e", ".")
+    session.run("bash", "-c", "cd doc && make html O=-W", external=True)
+
+
+@nox.session
+def tests(session: "nox.session") -> None:
+    session_id = "tests_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def tests_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "tests_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def tests_version(session: "nox.session", version: str) -> None:
+    session.install(
+        "jinja2",
+        "lxml",
+        "pygments==2.7.4",
+        "pytest",
+        "cmake",
+        "yaxmldiff",
+    )
+    coverage_args = []
+    if os.environ.get("USE_COVERAGE") == "true":
+        session.install("pytest-cov")
+        coverage_args = ["--cov=gcovr", "--cov-branch"]
+    session.install("-e", ".")
+    set_environment(session, version)
+    session.log("Print tool versions")
+    session.run("python", "--version")
+    session.run(session.env["CC"], "--version", external=True)
+    session.run(session.env["CXX"], "--version", external=True)
+    session.run(
+        session.env["GCOV"].replace(" gcov", "")
+        if "llvm-cov" in session.env["GCOV"]
+        else session.env["GCOV"],
```

Comment:
What does this do? I don't understand. Is this trying to un-shellquote the command? In `gcov.py`, we use `shlex.split()` for that purpose. Alternatively, we could yolo it and `run("sh", "-c", f"{session.env['GCOV']} --version")`

Reply:
Is the result of `llvm-cov gcov --version` the same as from `gcov --version`. If yes, than this isn't needed. I'll check this.